Client/Frontend/Browser/Tab.swift

```diff
@@ -256,6 +256,32 @@ class Tab: NSObject {
         contentScriptManager.helpers.removeAll()
     }
     
+    func clearHistory(config: WKWebViewConfiguration) {
+        guard let webView = webView,
+              let tabID = id  else {
+            return
+        }
+        
+        // Remove the tab history from saved tabs
+        TabMO.removeHistory(with: tabID)
+
+        /*
+         * Clear selector is used on WKWebView backForwardList because backForwardList list is only exposed with a getter
+         * and this method Removes all items except the current one in the tab list so when another url is added it will add the list properly
+         * This approach is chosen to achieve removing tab history in the event of removing  browser history
+         * Best way perform this is to clear the backforward list and in our case there is no drawback to clear the list
+         * And alternative would be to reload webpages which will be costly and also can cause unexpected results
+         */
+        let argument: [Any] = ["_c", "lea", "r"]
+
+        let method = argument.compactMap { $0 as? String }.joined()
```

Comment:
PR Looks good. This MIGHT not work if the optimizer is smart. If it is, this will get optimized back into a single string `_clear` and then get flagged by the automatic review for private methods :D 
Typically I base-64 encode the string and then decode it at runtime. But for now, leave it and let's see. Just adding a comment in case it does get rekt.

Reply:
Actually I added extra [obfuscation](https://github.com/brave/brave-ios/pull/3089#discussion_r531745554) which includes base-64 encode and unicode encoding but we decided it is unnecessary.
But like you said let's see, we can put it back.